Cubical/ZCohomology/RingStructure/CohomologyRing.agda

```diff
@@ -141,36 +87,14 @@ module CupRingProperties (A : Type ℓ) where
   -^-base : {k : ℕ} → (n m : ℕ) → (a : coHom k A) → (-^ n · m) (base k a) ≡ base k ((-ₕ^ n · m) a)
   -^-base n m a = -^-gen-base n m (evenOrOdd n) (evenOrOdd m) a
 
-
-module _ (A : Type ℓ) where
-  open intermediate-def renaming (H* to H*')
-  open CupRingProperties A
-  open AbGroupStr (snd (H*AbGr A))
-  open AbGroupTheory (H*AbGr A)
-
-  H*R : Ring ℓ
-  fst H*R = H*' A
-  RingStr.0r (snd H*R) = 0g
-  RingStr.1r (snd H*R) = 1H*
-  RingStr._+_ (snd H*R) = _+_
-  RingStr._·_ (snd H*R) = _cup_
-  RingStr.- snd H*R = -_
-  RingStr.isRing (snd H*R) = makeIsRing is-set
-                                        +Assoc +IdR +InvR +Comm
-                                        cupAssoc cupIdR cupIdL cupDistR cupDistL
-
-
-  H* : Type ℓ
-  H* = fst H*R
-
   gradCommRing : (n m : ℕ) → (a : coHom n A) → (b : coHom m A) →
                  (base n a) cup (base m b) ≡ (-^ n · m) ((base m b) cup (base n a))
   gradCommRing n m a b = base (n +' m) (a ⌣ b)
                                  ≡⟨ cong (base (n +' m)) (gradedComm-⌣ n m a b) ⟩
                         base (n +' m) ((-ₕ^ n · m) (subst (λ n₁ → coHom n₁ A) (+'-comm m n) (b ⌣ a)))
                                  ≡⟨ sym (-^-base n m (subst (λ k → coHom k A) (+'-comm m n) (b ⌣ a))) ⟩
                         (-^ n · m)  (base (n +' m) (subst (λ k → coHom k A) (+'-comm m n) (b ⌣ a)))
-                                 ≡⟨ cong (-^ n · m) (sym (constSubstCommSlice (λ k → coHom k A) H* base (+'-comm m n) (b ⌣ a))) ⟩
+                                 ≡⟨ cong (-^ n · m) (sym (constSubstCommSlice (λ k → coHom k A) (H* A) base (+'-comm m n) (b ⌣ a))) ⟩
                          (-^ n · m) (base (m +' n) (b ⌣ a)) ∎
 
 
```

Comment:
I didn't generalize it because you need that the index is the same definitionally which is annoying.
May as well keep it this way, as the only two cases I need it are already in the library.